packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -321,17 +322,51 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
 
     // Queue promises to fetch independent data.
     const nonceFetchPromise = (async () => {
-      const _nonceSpace = buildUseropDto?.nonceOptions?.nonceKey ?? 0;
-      const nonce = await this.getNonce(_nonceSpace);
+      let nonce = BigNumber.from(0);
+      try {
+        if (buildUseropDto?.nonceOptions?.nonceOverride) {
+          nonce = BigNumber.from(buildUseropDto?.nonceOptions?.nonceOverride);
+        } else {
+          const _nonceSpace = buildUseropDto?.nonceOptions?.nonceKey ?? 0;
+          nonce = await this.getNonce(_nonceSpace);
+        }
+      } catch (error) {
+        // Not throwing this error as nonce would be 0 if this.getNonce() throw exception, which is expected flow for undeployed account
+      }
       return nonce;
     })();
     const initCodeFetchPromise = this.getInitCode();
     const dummySignatureFetchPromise = this.getDummySignature(buildUseropDto?.params);
 
+    const gasFeeValues = {
```

Comment:
Remove thes inner functions to outside of this buildUserOp method and rename methods to

getGasFeeValues
getNonce

Reply:
we already have a `getNonce` in SmartAccountV2, this method was added just to call that method async in anonmyouse function so the `getGasFeeValues` and `getNonce` are fetched parallely.

Reply:
Should I rename it to something else then?